.circleci/config.yml

```diff
@@ -0,0 +1,201 @@
+
+  - &restore-cache-android
+    keys:
+      - v1-android-dependencies-{{ .Branch }}-{{ checksum "ReactAndroid/build.gradle" }}
+      # Fallback in case checksum fails
+      - v1-android-dependencies-{{ .Branch }}-
+  - &save-cache-android
+    paths:
+     - "ReactAndroid/build/downloads"
+    key: v1-android-dependencies-{{ .Branch }}-{{ checksum "ReactAndroid/build.gradle" }}
+
+  - &restore-cache-buck
+    keys:
+# TODO: Pick file to checksum against
+      - v1-buck-dependencies-{{ .Branch }}
+  - &save-cache-buck
+    paths:
+     - "/home/circleci/buck"
+    key: v1-buck-dependencies-{{ .Branch }}
+
+version: 2
+jobs:
+  test-node-8:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:8
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run: |
+          npm run lint
+          npm run flow -- check
+          npm test -- --maxWorkers=1
+
+  test-node-6:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:6.11.0
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run: |
+          npm run lint
+          npm run flow -- check
+          npm test -- --maxWorkers=1
+
+  test-node-4:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:4.8.4
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run: |
+          npm run lint
+          npm run flow -- check
+          npm test -- --maxWorkers=1
+
+  build-js-bundle:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:8
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run:
+          name: Building JS bundle for instrumentation tests
+          command: node local-cli/cli.js bundle --max-workers 1 --platform android --dev true --entry-file ReactAndroid/src/androidTest/js/TestBundle.js --bundle-output ReactAndroid/src/androidTest/assets/AndroidTestBundle.js
+      - persist_to_workspace:
+          root: ReactAndroid/src/androidTest/assets/
+          paths:
+            - AndroidTestBundle.js
+
+  test-android:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/openjdk:8-jdk
+    environment:
+      - TERM: "dumb"
+      - ADB_INSTALL_TIMEOUT: 10
+      - GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx512m -XX:+HeapDumpOnOutOfMemoryError"'
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ReactAndroid/src/androidTest/assets/
+      - run: |
+          if [[ ! -e ReactAndroid/src/androidTest/assets/AndroidTestBundle.js ]]; then
+            echo "JS bundle missing, verify build-js-bundle step"; exit 1
+          else
+            echo "JS bundle found.";
+          fi
+# dependencies
+      - run:
+          name: Setting up Buck and Android dependencies
+          command: |
+            if [[ ! -e /home/circleci/buck ]]; then
```

Comment:
Small suggestion - how about using "~/buck" just to avoid thinking as much about how CircleCI does homedirs?

Reply:
Done.